File: src/lesson005/java/BankAccountTest.java
package lesson005.java;

public class BankAccountTest {
    public static void main(String[] args) {
        BankAccount myAccount = new BankAccount();

        myAccount.id = 1;
        myAccount.name = "Mike";
        myAccount.balance = 1500.12;

        myAccount.cashIn(150.55);
        myAccount.cashOut(1500);
        myAccount.cashOut(5000);
    }
}

class BankAccount{
    int id = 0;
    String name = "NoName";
    double balance = 0.0;

    public void cashIn(double cashSum){
        if(cashSum>=0.0) {
            balance += cashSum;
            System.out.println("Cash in successful. Now is "+String.format("%.2f", balance));
        } else System.out.println("Incorrect cash value.");
    }

    public void cashOut(double cashSum){
        if(cashSum<=balance){
            balance-=cashSum;
            System.out.println("Cash out successful. Now is "+String.format("%.2f", balance));
        } else System.out.println("You enter sum more than you balance. Now is "+String.format("%.2f", balance));
    }
}

File: src/lesson005/kotlin/BankAccountTest.kt
package lesson005.kotlin

import java.text.NumberFormat
import java.util.*
private val FORMAT_US: NumberFormat = NumberFormat.getCurrencyInstance(Locale.US)


fun main(args: Array<String>) {
    val account = BankAccount(154, "Mike", 1500.00)
    println("Hello, ${account.name}. You id is ${account.id}. You balance is ${FORMAT_US.format(account.balance)}.")
    account.cashIn(150.00)
    account.cashOut(1500.00)
    account.cashOut(5000.00)

}

class BankAccount(
    var id: Int,
    var name: String,
    var balance: Double
){
    fun cashIn(cashSum: Double) = if(cashSum>0.0){
        balance+=cashSum
        println("Cash in successful. Now is ${FORMAT_US.format(balance)}")
    } else println("Incorrect cash value.")

    fun cashOut(cashSum: Double) = if (cashSum <= balance) {
        balance -= cashSum
        println("Cash out successful. Now is ${FORMAT_US.format(balance)}")
    } else println("You enter sum more than you balance. Now is ${FORMAT_US.format(balance)}")
}
